config/uaa/enable-automount-service-account-token-for-kind.yml

```diff
@@ -0,0 +1,13 @@
+#@ load("@ytt:data", "data")
```

Comment:
For some historical context here (from my view, which may be missing lots of data and perspectives):

1. We (cf-k8s-networking) started the work to support Kind with Istio 1.6.4 a couple months ago. In our original design, we were planning to use data values to thread this info through. We had PRs and conversations with the eirini and uaa teams to add the necessary configuration to their subcomponents before we could tie it all together at the cf-for-k8s level.
2. For several reasons, the Istio 1.6.4 work went on longer than we expected. By the time we picked up this work again, I think the context of the original plan of using data values was lost. We instead went down the path of overlays at this point, but we had some vestigial pieces showing our original approach. @davewalter noticed one of these [while reviewing the PR](https://github.com/cloudfoundry/cf-for-k8s/pull/349#discussion_r466572563) that ultimately got Istio 1.6.4 into cf-for-k8s.
3. When we were recently preparing the PR for bumping Istio, we noticed that this change in direction had happened, probably unintentionally, but we wanted to first focus on getting Istio bumped at all before considering whether we wanted to move back to the data values approach.
4. Given all that, this file here is probably overly complicated compared to just passing a data value through when we generate the uaa library's data values. That was our original plan and the reason for [this PR](https://github.com/cloudfoundry/uaa/pull/1326) to UAA, and we're happy to help simplify some of this.

Sorry the verbosity for a relatively simple thing. I just wanted to have somewhere to capture what our intentions were and how we got here :).

Reply:
Thanks for the context @jenspinney! I think this will ultimately be decided by the interface that @Syerram wanted to expose to the operator. Personally, I do like having a kind-specific configuration that turns on everything required to deploy.